client/src/components/CreateSession.jsx

```diff
@@ -41,10 +45,42 @@ class CreateSession extends React.Component {
     // };
     // const response = await axios(config);
 
-    // Temporary response variable while we use dummy JS file
-    const response = await axios.get('http://localhost:8080/users/testGeo');
-    const restaurants = response.data.data;
-    this.setState({ restaurants });
+    try {
+      this.getUserLocation();
+      // Temporary response variable while we use dummy JS file
+      const response = await axios.get('http://localhost:8080/users/testGeo');
+      const restaurants = response.data.data;
+      this.setState({ restaurants });
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  getPosition (options) {
+    return new Promise((resolve, reject) => {
+      navigator.geolocation.getCurrentPosition(resolve, reject, options);
+    });
+  }
+
+  async getUserLocation () {
+    const options = {
+      enableHighAccuracy: false,
+      timeout: 5000,
+      maximumAge: 0
+    };
+
+    try {
+      const pos = await this.getPosition(options);
+      const hostGeo = {
+        lat: pos.coords.latitude,
+        lon: pos.coords.longitude
+      };
+      this.props.setTopLevelState('hostGeo', hostGeo);
+      this.setState({ geolocationProcessComplete: true });
+    } catch (err) {
+      console.error(err);
+      this.setState({ geolocationProcessComplete: true });
+    }
```

Comment:
nit, you could add this line to a `finally` block to make it more DRY

```suggestion
      this.props.setTopLevelState('hostGeo', hostGeo);
    } catch (err) {
      console.error(err);
    } finally {
      this.setState({ geolocationProcessComplete: true });
    }
```

Reply:
Ah great catch! Updated